### PR TITLE
Add normalize script for IN arcgis data

### LINKED ISSUE #57: normalize arcgis from Indiana
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/main/vaccine_feed_ingest/schema/schema.py).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch ia/arcgis
    poetry run vaccine-feed-ingest parse ia/arcgis
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize ia/arcgis
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by Hain0**
I copied from the sample AK normalize script and adjusted for the IN input, also see #111. Added now because I'm going to sleep soon.

Resolves #57

### CODE REVIEW

File: vaccine_feed_ingest/runners/in/arcgis/normalize.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    data_id = site["attributes"]["GlobalId"]
+
+    # Could parse these from directory traversal, but do not for now to avoid
+    # accidental mutation.
+    site = "arcgis"
+    runner = "in"
+
+    # Could parse these from the input file name, but do not for now to avoid
+    # accidental mutation.
+    arcgis = "97135bbb1bec488e9717aca061c03e41"
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/in/arcgis/normalize.py:40]: Be sure to update this for this state!

File: vaccine_feed_ingest/runners/in/arcgis/normalize.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+
+def _get_inventory(site: dict) -> Optional[List[schema.Vaccine]]:
+    vaccines_field = site["attributes"]["Vaccine_Type"].lower().split(",")
+
+    potentials = {
+        "Pfizer": schema.Vaccine(vaccine="pfizer"),
+        "Moderna": schema.Vaccine(vaccine="moderna"),
+        "Johnson & Johnson": schema.Vaccine(vaccine="janssen"),
+        "FRPP": schema.Vaccine(vaccine="frpp"),
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/in/arcgis/normalize.py:53]: "FRPP stands for ["Federal Retail Pharmacy Partnership"](https://www.cdc.gov/vaccines/covid-19/retail-pharmacy-program/index.html), and is not a vaccine type.

If a row has this data, let's avoid setting the type of the vaccine.

File: vaccine_feed_ingest/runners/in/arcgis/normalize.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        inventory=_get_inventory(site),
+        access=None,
+        parent_organization=None,
+        links=None,
+        notes=_get_notes(site),
+        active=None,
+        source=schema.Source(
+            source="arcgis",
+            id=site["attributes"]["GlobalID"],
+            fetched_from_uri="https://services1.arcgis.com/WzFsmainVTuD5KML/ArcGIS/rest/services/COVID19_Vaccine_Site_Survey_API/FeatureServer/0",  # noqa: E501
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/in/arcgis/normalize.py:139]: Please set this to `https://experience.arcgis.com/experience/24159814f1dd4f69b6c22e7e87bca65b`

File: vaccine_feed_ingest/runners/in/arcgis/normalize.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+def _get_contacts(site: dict) -> Optional[List[schema.Contact]]:
+    contacts = []
+    if site["attributes"]["Site_Phone"]:
+        sourcePhone = re.sub("[^0-9]", "", site["attributes"]["Site_Phone"])
+        if len(sourcePhone) == 11:
+            sourcePhone = sourcePhone[1:]
+        phone = f"({sourcePhone[0:3]}) {sourcePhone[3:6]}-{sourcePhone[6:]}"
+        contacts.append(schema.Contact(phone=phone))
+
+    if site["attributes"]["Site_Zotec_Link"]:
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/in/arcgis/normalize.py:79]: It looks like there are additionally sometimes links in the `Promote_Name` field. Is it possible to extract those?

File: vaccine_feed_ingest/runners/in/arcgis/normalize.py [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+        contacts.append(schema.Contact(other=site["attributes"]["Site_Location_Info"]))
+
+    if len(contacts) > 0:
+        return contacts
+
+    return None
+
+
+def _get_notes(site: dict) -> Optional[List[str]]:
+    if site["attributes"]["Site_Special_Inst"]:
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/in/arcgis/normalize.py:92]: It looks like there may also be notes in `Site_Location_Info`. Could we add those?

**Review by eliblock — COMMENTED** [review]
Thank you for starting on this normalizer! It looks like there may be a bit more data that we can extract 😄.

As a reminder, you should be able to run your code locally by running `poetry run vaccine-feed-ingest all-stages in/arcgis` (and, after the first run, you can use simply `poetry run vaccine-feed-ingest normalize in/arcgis`). Please give that a try so you can see the output data locally!

**Review by eliblock — CHANGES_REQUESTED** [review]
Thank you for the fixes from the first comments!

It looks to me like this normalizer doesn't run quite yet due to some mis-matched attributes. Could you give it a try (`poetry run vaccine-feed-ingest all-stages in/arcgis`) to make sure the data processes as you expect?

File: vaccine_feed_ingest/runners/in/arcgis/normalize.py [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    arcgis = "46630b2520ce44a68a9f42f8343d3518"
+    layer = 0
+
+    return f"{runner}:{site}:{arcgis}_{layer}:{data_id}"
+
+
+def _get_inventory(site: dict) -> Optional[List[schema.Vaccine]]:
+    vaccines_field = site["attributes"]["Vaccine_Type"].lower().split(",")
+
+    potentials = {
```
Comment (tmb) [inline vaccine_feed_ingest/runners/in/arcgis/normalize.py:49]: Can you use the vaccine name constants specified on this wiki page? https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema

Thank you!

Comment (eliblock) [reply vaccine_feed_ingest/runners/in/arcgis/normalize.py:49]: Sorry about that - these changed last night to match another system, and I forgot to update my review here to include the new values. My bad!

**Review by eliblock — APPROVED** [review]
I made some small fixes to get this running (e.g., `GlobalID` rather than `GlobalId`), and also:
* updated vaccine inventory calculations (it previously did not match most vaccines)
* updated address parsing (this update left _fewer_ parsed addresses, but removed some buggy ones that appeared with the original address. I encourage a followup PR to improve the address match rate - I'll open an issue for it!)

With that, I'm going to merge this! Thank you for bringing these sites online!

**Pull request merged**